engine/executor/logic_plan.go

```diff
@@ -4316,3 +4324,79 @@ func (p *LogicalBinOp) Digest() string {
 	p.digestName = encoding.MarshalUint64(p.digestName, p.right.ID())
 	return string(p.digestName)
 }
+
+type LogicalPromSubquery struct {
+	input hybridqp.QueryNode
+	Call  *influxql.PromSubCall
+	LogicalPlanBase
+}
+
+func NewLogicalPromSubquery(input hybridqp.QueryNode, schema hybridqp.Catalog, Call *influxql.PromSubCall) *LogicalPromSubquery {
+	project := &LogicalPromSubquery{
+		input: input,
+		Call:  Call,
+		LogicalPlanBase: LogicalPlanBase{
+			id:     hybridqp.GenerateNodeId(),
+			schema: schema,
+			rt:     nil,
+			ops:    nil,
+		},
+	}
+	project.init()
+	return project
+}
+
+// impl me
+func (p *LogicalPromSubquery) New(inputs []hybridqp.QueryNode, schema hybridqp.Catalog, eTrait []hybridqp.Trait) hybridqp.QueryNode {
+	return nil
+}
+
+func (p *LogicalPromSubquery) DeriveOperations() {
+	p.init()
+}
+
+func (p *LogicalPromSubquery) init() {
+	p.InitRef(p.input)
+}
+
+func (p *LogicalPromSubquery) Clone() hybridqp.QueryNode {
+	clone := &LogicalPromSubquery{}
+	*clone = *p
+	clone.id = hybridqp.GenerateNodeId()
+	return clone
+}
+
+func (p *LogicalPromSubquery) Children() []hybridqp.QueryNode {
+	return []hybridqp.QueryNode{p.input}
+}
+
+func (p *LogicalPromSubquery) ReplaceChildren(children []hybridqp.QueryNode) {
+	p.input = children[0]
```

Comment:
Is it possible for children to be nil?

Reply:
The child operator of promsubquery is not nil.